mllib/src/main/scala/org/apache/spark/ml/feature/MaxAbsScaler.scala

```diff
@@ -70,14 +67,40 @@ class MaxAbsScaler @Since("2.0.0") (@Since("2.0.0") override val uid: String)
   @Since("2.0.0")
   override def fit(dataset: Dataset[_]): MaxAbsScalerModel = {
     transformSchema(dataset.schema, logging = true)
-    val input: RDD[OldVector] = dataset.select($(inputCol)).rdd.map {
-      case Row(v: Vector) => OldVectors.fromML(v)
-    }
-    val summary = Statistics.colStats(input)
```

Comment:
is it the call to colStats and vector conversion that was so inefficient?  do you have any performance numbers to justify the change, since it does make the code more complicated.